include/envoy/http/codec.h

```diff
@@ -146,6 +146,16 @@ class ConnectionCallbacks {
    */
   virtual void onGoAway() PURE;
 };
+/**
+ * HTTP/1.* Codec settings
+ */
+
+struct Http1Settings {
+  // Enable codec to handle CONNECT requests. This will enable forward/explicit proxy support on TLS connections
+  bool allow_connect_{false} ;
+  // Enable codec to parse absolute uris. This enables forward/explicit proxy support for non TLS traffic
+  bool allow_absolute_url_{false};
```

Comment:
To your other point, there aren't any non-proprietary clients that use absolute-url in the request, but I don't see a reason to preemptively break it.